features/networking/ovn.feature

```diff
@@ -465,3 +465,61 @@ Feature: OVN related networking scenarios
     Then the step should succeed
     # the iface-id should be quoted if it starts with a digit
     And the output should contain "<%= project.name %>"
+
+
+# @author huirwang@redhat.com
+# @case_id OCP-38132
+@admin
+@destructive
+Scenario: Should no intermittent packet drop from pod to pod after change hostname
+  Given I store the schedulable workers in the :nodes clipboard
+  Given admin uses the "openshift-ovn-kubernetes" project
+  And I store the hostname from external ids in the :original_hostname clipboard on the "<%= cb.nodes[0].name %>" node
+  Then the step should succeed
+  And I store the "short" hostname in the :short_hostname clipboard for the "<%= cb.nodes[0].name %>" node
+  Then the step should succeed
+
+  # Configure short hostname to the external_ids for the ovn node
+  When I set "<%= cb.short_hostname %>" hostname to external ids on the "<%= cb.nodes[0].name %>" node
+  Then the step should succeed
+  Given I register clean-up steps:
+  """
+  When I set "<%= cb.original_hostname%>" hostname to external ids on the "<%= cb.nodes[0].name %>" node
+  Then the step should succeed
+  """
+
+  # Check short hostname was set
+  Given I store the ovnkube-master "north" leader pod in the clipboard
+  Given I wait up to 15 seconds for the steps to pass:
+  """
+  And admin executes on the pod "northd" container:
+    | bash | -c | ovn-sbctl --no-leader-only list chassis \|grep "<%= cb.short_hostname %>" |
+  Then the step should succeed
+  And the output should contain:
+    | <%= cb.short_hostname %> |
+  And the output should not contain:
+    | <%= cb.original_hostname%> |
```

Comment:
Doesn't seem to work on vSphere.  The external hostname is the same as the shortname.

```
external_ids        : {hostname=control-plane-0, ovn-bridge-mappings="physnet:br-ex", ovn-encap-ip="172.31.248.16", ovn-encap-type=geneve, ovn-monitor-all="true", ovn-openflow-probe-interval="180", ovn-remote="ssl:172.31.248.16:9642,ssl:172.31.248.48:9642,ssl:172.31.248.86:9642", ovn-remote-probe-interval="30000", rundir="/var/run/openvswitch", system-id="233f3038-086f-4a2e-8e03-1601e7f08a78"}
external_ids        : {hostname=compute-1, ovn-bridge-mappings="physnet:br-ex", ovn-encap-ip="172.31.248.54", ovn-encap-type=geneve, ovn-monitor-all="true", ovn-openflow-probe-interval="180", ovn-remote="ssl:172.31.248.16:9642,ssl:172.31.248.48:9642,ssl:172.31.248.86:9642", ovn-remote-probe-interval="30000", rundir="/var/run/openvswitch", system-id="1e994f24-bc20-43d7-804d-ef88e1ccd591"}
external_ids        : {hostname=control-plane-1, ovn-bridge-mappings="physnet:br-ex", ovn-encap-ip="172.31.248.86", ovn-encap-type=geneve, ovn-monitor-all="true", ovn-openflow-probe-interval="180", ovn-remote="ssl:172.31.248.16:9642,ssl:172.31.248.48:9642,ssl:172.31.248.86:9642", ovn-remote-probe-interval="30000", rundir="/var/run/openvswitch", system-id="1f28920a-e538-42e3-8777-c8170e181f15"}
external_ids        : {hostname=compute-0, ovn-bridge-mappings="physnet:br-ex", ovn-encap-ip="172.31.248.19", ovn-encap-type=geneve, ovn-monitor-all="true", ovn-openflow-probe-interval="180", ovn-remote="ssl:172.31.248.16:9642,ssl:172.31.248.48:9642,ssl:172.31.248.86:9642", ovn-remote-probe-interval="30000", rundir="/var/run/openvswitch", system-id="c2e44006-4437-4cfe-9c78-a090bd060087"}
external_ids        : {hostname=control-plane-2, ovn-bridge-mappings="physnet:br-ex", ovn-encap-ip="172.31.248.48", ovn-encap-type=geneve, ovn-monitor-all="true", ovn-openflow-probe-interval="180", ovn-remote="ssl:172.31.248.16:9642,ssl:172.31.248.48:9642,ssl:172.31.248.86:9642", ovn-remote-probe-interval="30000", rundir="/var/run/openvswitch", system-id="cf3944a9-03b4-4ff3-89fe-da5c4d04a68b"}
external_ids        : {hostname=compute-2, ovn-bridge-mappings="physnet:br-ex", ovn-encap-ip="172.31.248.74", ovn-encap-type=geneve, ovn-monitor-all="true", ovn-openflow-probe-interval="180", ovn-remote="ssl:172.31.248.16:9642,ssl:172.31.248.48:9642,ssl:172.31.248.86:9642", ovn-remote-probe-interval="30000", rundir="/var/run/openvswitch", system-id="402d2f1e-b021-45e1-b4b9-f1f9514e2c91"}
```

Reply:
```
oc exec ovnkube-master-pm274  --kubeconfig=ocp4_admin.kubeconfig -n openshift-ovn-kubernetes  --container=northd -i -- bash -c ovn-sbctl\ --no-leader-only\ list\ chassis\ \|grep\ \"compute-0\"

hostname            : compute-0

STDERR:
E0420 08:14:14.136111 3394046 v2.go:105] write tcp 172.16.16.104:54570->54.189.75.126:3128: write: broken pipe


[12:14:14] INFO> Exit Status: 0

RuntimeError: pattern found: compute-0
```

Reply:
Thanks Ross for reviewing it. Actually this case is from bug https://bugzilla.redhat.com/show_bug.cgi?id=1908921 and from comments https://bugzilla.redhat.com/show_bug.cgi?id=1908921#c17, we can see the issue was reproduced on AWS and we also followed that suggestion to verify the bug.  I have filtered that case in Polarion for AWS platform.